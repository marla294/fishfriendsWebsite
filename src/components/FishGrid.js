import React from "react";
import PropTypes from "prop-types";
import Fish from "./Fish";

class FishGrid extends React.Component {
	renderFish = () => {
		return this.props.fishes.map(fish => (
			<Fish key={fish.Id} name={fish.Name} />
		));
	};

	render() {
		return <div>{this.renderFish()}</div>;
	}
}

FishGrid.propTypes = {
	fishes: PropTypes.array.isRequired
};

export default FishGrid;
