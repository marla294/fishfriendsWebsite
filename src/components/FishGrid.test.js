import React from "react";
import FishGrid from "./FishGrid";
import renderer from "react-test-renderer";

test("Fish appear in FishGrid", () => {
	const fishes = [
		{ Id: 1, Name: "dwarf angels", Info: [] },
		{ Id: 2, Name: "large angels", Info: [] }
	];

	const component = renderer.create(<FishGrid fishes={fishes} />);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
