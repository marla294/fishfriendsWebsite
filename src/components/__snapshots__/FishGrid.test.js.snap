// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fish appear in FishGrid 1`] = `
<div>
  <div
    className="fish "
    onClick={[Function]}
  >
    <img
      alt="fishPic"
      src="/images/fish/dwarf angels.png"
    />
    <div
      className="name"
    >
      Dwarf angels
    </div>
  </div>
  <div
    className="fish "
    onClick={[Function]}
  >
    <img
      alt="fishPic"
      src="/images/fish/large angels.png"
    />
    <div
      className="name"
    >
      Large angels
    </div>
  </div>
</div>
`;
